NorthstarDLL/plugins/plugins.cpp

```diff
@@ -1,422 +1,280 @@
 }
 
-void initGameState()
-{
-	// Initalize the Slim Reader / Writer locks
-	InitializeSRWLock(&gameStateLock);
-	InitializeSRWLock(&serverInfoLock);
-	InitializeSRWLock(&playerInfoLock);
-
-	gameStateExport.getGameStateChar = &getGameStateChar;
-	gameStateExport.getGameStateInt = &getGameStateInt;
-	gameStateExport.getGameStateBool = &getGameStateBool;
-
-	serverInfoExport.getServerInfoChar = &getServerInfoChar;
-	serverInfoExport.getServerInfoInt = &getServerInfoInt;
-	serverInfoExport.getServerInfoBool = &getServerInfoBool;
-
-	playerInfoExport.getPlayerInfoChar = &getPlayerInfoChar;
-	playerInfoExport.getPlayerInfoInt = &getPlayerInfoInt;
-	playerInfoExport.getPlayerInfoBool = &getPlayerInfoBool;
-
-	serverInfo.id = "";
-	serverInfo.name = "";
-	serverInfo.description = "";
-	serverInfo.password = "";
-	serverInfo.maxPlayers = 0;
-	gameState.connected = false;
-	gameState.loading = false;
-	gameState.map = "";
-	gameState.mapDisplayName = "";
-	gameState.playlist = "";
-	gameState.playlistDisplayName = "";
-	gameState.players = 0;
-
-	playerInfo.uid = 123;
-}
-
-// string gamemode, string gamemodeName, string map, string mapName, bool connected, bool loading
-SQRESULT SQ_UpdateGameStateUI(HSquirrelVM* sqvm)
-{
-	AcquireSRWLockExclusive(&gameStateLock);
-	gameState.map = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 1);
-	gameState.mapDisplayName = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 2);
-	gameState.playlist = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 3);
-	gameState.playlistDisplayName = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 4);
-	gameState.connected = g_pSquirrel<ScriptContext::UI>->getbool(sqvm, 5);
-	gameState.loading = g_pSquirrel<ScriptContext::UI>->getbool(sqvm, 6);
-	ReleaseSRWLockExclusive(&gameStateLock);
-	return SQRESULT_NOTNULL;
-}
-
-// int playerCount, int outScore, int secondHighestScore, int highestScore, bool roundBased, int scoreLimit
-SQRESULT SQ_UpdateGameStateClient(HSquirrelVM* sqvm)
+EXPORT void PLUGIN_LOG(LogMsg* msg)
 {
-	AcquireSRWLockExclusive(&gameStateLock);
-	AcquireSRWLockExclusive(&serverInfoLock);
-	gameState.players = g_pSquirrel<ScriptContext::CLIENT>->getinteger(sqvm, 1);
-	serverInfo.maxPlayers = g_pSquirrel<ScriptContext::CLIENT>->getinteger(sqvm, 2);
-	gameState.ourScore = g_pSquirrel<ScriptContext::CLIENT>->getinteger(sqvm, 3);
-	gameState.secondHighestScore = g_pSquirrel<ScriptContext::CLIENT>->getinteger(sqvm, 4);
-	gameState.highestScore = g_pSquirrel<ScriptContext::CLIENT>->getinteger(sqvm, 5);
-	serverInfo.roundBased = g_pSquirrel<ScriptContext::CLIENT>->getbool(sqvm, 6);
-	serverInfo.scoreLimit = g_pSquirrel<ScriptContext::CLIENT>->getbool(sqvm, 7);
-	ReleaseSRWLockExclusive(&gameStateLock);
-	ReleaseSRWLockExclusive(&serverInfoLock);
-	return SQRESULT_NOTNULL;
+	spdlog::source_loc src {};
+	src.filename = msg->source.file;
+	src.funcname = msg->source.func;
+	src.line = msg->source.line;
+	auto&& logger = g_pPluginManager->m_vLoadedPlugins[msg->pluginHandle].logger;
```

Comment:
the logger crashes here if a plugin isn't loaded correctly 
and a plugin after it calls the log function.

since the plugin handle would be larger than the vector